src/Symfony/Component/Validator/Constraints/ExpressionValidator.php

```diff
@@ -27,6 +27,7 @@ class ExpressionValidator extends ConstraintValidator
     public function __construct(ExpressionLanguage $expressionLanguage = null)
     {
         $this->expressionLanguage = $expressionLanguage;
+        $this->expressionLanguage?->registerProvider(new ExpressionLanguageProvider());
```

Comment:
We shouldn't mutate the expression language we receive in the constructor, it would lead to unexpected behavior if the provided expression language is used elsewhere.